src/ex4/server/clients/pokemon_client.js

```diff
@@ -0,0 +1,38 @@
+import axios from "axios";
+const url = "https://pokeapi.co/api/v2/pokemon";
+let requestsArray = [];
+
+export async function fetchPokemon(pokemonId) {
+  const date = new Date();
+  let curretHour = parseInt(date.getHours());
+  let curretMinute = parseInt(date.getMinutes());
+  const isThisRequestExist = [...requestsArray].some((item) => {
+    if (item.id === pokemonId) {
+      if (item.hour === curretHour) {
+        if (curretMinute - item.minute < 2) {
```

Comment:
this multiple if is a bit unclear
can refactor to one with &&
or the most clear why in my opinion is to export to a function `isCurrentRequestData`